sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/AvoidExcessiveInheritance.cs

```diff
@@ -51,6 +50,30 @@ public class AvoidExcessiveInheritance : ParameterLoadingDiagnosticAnalyzer
             defaultValue: MaximumDepthDefaultValue)]
         public int MaximumDepth { get; set; } = MaximumDepthDefaultValue;
 
+        private ICollection<Regex> filteredClassesRegex = new List<Regex>();
+
+        private const string FilteredClassesDefaultValue = "";
+        private string filteredClasses = FilteredClassesDefaultValue;
+        [RuleParameter(
+            key: "filteredClasses",
+            type: PropertyType.String,
+            description: "Classes to be filtered out of the count of inheritance. Depth counting will stop when a filtered class is reached. (String)",
+            defaultValue: FilteredClassesDefaultValue)]
+        public string FilteredClasses
+        {
+            get
+            {
+                return filteredClasses;
+            }
+            set
+            {
+                filteredClasses = value;
+                filteredClassesRegex = filteredClasses.Split(',')
+                    .Select(s => new Regex(WilcardToRegular(s), RegexOptions.Compiled))

```

Comment:
You could put new Regex() into WildcardToRegular and this code will become:

    ....
    .Select(WildcardToRegex)
    .ToList()